ops/framework.py

```diff
@@ -32,6 +32,26 @@
 from ops import charm
 from ops.storage import NoSnapshotError, SQLiteStorage
 
+if typing.TYPE_CHECKING:
+    from pathlib import Path
+
+    from typing_extensions import Protocol, Type
+
+    from ops.charm import CharmMeta
+    from ops.model import Model
+
+    class _Serializable(Protocol):
+        handle = None  # type: Handle
+        def snapshot(self) -> dict: ...  # noqa: E704
+        def restore(self, snapshot: dict) -> "Object": ...  # noqa: E704
+
+    _ObjectType = typing.TypeVar("_ObjectType", bound="Object")
+    _EventType = typing.TypeVar("_EventType", bound=Type["EventBase"])
+    _Observer = typing.Callable[[typing.Any], None]
+    _Path = _Kind = str
+    _ParentType = typing.Union["Handle", _ObjectType]
```

Comment:
Why not `_HandleType`?

Reply:
Maybe you could throw in some doc comments describing what this type represents since it seems to be a bit tricky to determine based on any of these names.